aten/src/ATen/native/quantized/cudnn/Conv.cpp

```diff
@@ -249,22 +261,46 @@ void raw_cudnn_convolution_forward_out(
     std::vector<void *> data_ptrs;
     std::vector<int64_t> uids;
     if (bias.has_value()) {
-      data_ptrs = std::vector<void *>{reinterpret_cast<int8_t*>(input.data_ptr()), conv_output.data_ptr(),
-                                  reinterpret_cast<int8_t*>(weight.data_ptr()),
-                                  requantize_multiplier_tensor.data_ptr(),
-                                  reinterpret_cast<int8_t*>(quantized_output.data_ptr()),
-                                  broadcasted_bias.value().data_ptr(),
-                                  bias_multiplier_tensor.value().data_ptr(),
-                                  after_scales_bias.value().data_ptr(),
-                                  after_add.value().data_ptr(),
-                                  };
-      uids = std::vector<int64_t>{'x', 'y', 'w', 's', 'r', 'b', 'c', 'd', 'e'};
+      if (kReluFused) {
+        data_ptrs = std::vector<void *>{reinterpret_cast<int8_t*>(input.data_ptr()), conv_output.data_ptr(),
+                                    reinterpret_cast<int8_t*>(weight.data_ptr()),
+                                    requantize_multiplier_tensor.data_ptr(),
+                                    reinterpret_cast<int8_t*>(quantized_output.data_ptr()),
+                                    broadcasted_bias.value().data_ptr(),
+                                    bias_multiplier_tensor.value().data_ptr(),
+                                    after_scales_bias.value().data_ptr(),
+                                    after_add.value().data_ptr(),
+                                    after_relu.value().data_ptr(),
+                                    };
+        uids = std::vector<int64_t>{'x', 'y', 'w', 's', 'r', 'b', 'c', 'd', 'e', 'f'};
+      } else {
+        data_ptrs = std::vector<void *>{reinterpret_cast<int8_t*>(input.data_ptr()), conv_output.data_ptr(),
+                                    reinterpret_cast<int8_t*>(weight.data_ptr()),
+                                    requantize_multiplier_tensor.data_ptr(),
+                                    reinterpret_cast<int8_t*>(quantized_output.data_ptr()),
+                                    broadcasted_bias.value().data_ptr(),
+                                    bias_multiplier_tensor.value().data_ptr(),
+                                    after_scales_bias.value().data_ptr(),
+                                    after_add.value().data_ptr(),
+                                    };
+        uids = std::vector<int64_t>{'x', 'y', 'w', 's', 'r', 'b', 'c', 'd', 'e'};
+      }
```

Comment:
can we try to eliminate the duplicated code here and below?

Reply:
I can, but I would have to use std::emplace_back to insert the remaining data pointers. I think it might be slightly slower because emplace_back may require a reallocation of the size exceeds the allocated space, so that's why I had it like this. Should we change it?

Reply:
actually i can avoid this issue with std::vector::reserve. i'll just reserve space for 10 pointers. so i'll make this change

Reply:
yes, reserve makes sense